### ISSUE TITLE
ipython.html.widgets.interact not working with figure.canvas.draw_idle() ?

### ISSUE THREAD

**Issue by LightServant** [ipython/ipython#7402]
Hello,

I am fairly new to ipython and programming. I've come to use it for post data processing in engineering and  love the python notebooks for documentation. 
I want to use interact function from ipython.html.widgets to select a data set and display it inline in the notebook in matplotlib object. 

This is a minimal example (without all the formatting and so on) of the code I use:

```python
%matplotlib nbagg
import matplotlib.pyplot as plt # Matlab like plotting
import numpy as np  #  Numeric & Array Handling
from IPython.html.widgets import interact

mydata = np.genfromtxt('mixg.csv', delimiter=',', dtype=float, names=True)
# Use of names implies data structure to be structured  in 1-dim array of tuples.
# Each Data-Set (column = tuple element) is accesible via name

# own names for columns
mydata.dtype.names=('time', r'-73dBm', r'-68dBm', r'-63dBm', r'-58dBm', r'-53dBm')

fig, ax = plt.subplots()
ax.plot(mydata['time'], mydata[r'-73dBm'])

plt.show()

def manipulate_plot(figure, data, selectionset, n_axis = 0, n_line = 0):
    def update_ydata(string):
        figure.axes[n_axis].lines[n_line].set_ydata(data[string])
        figure.canvas.draw_idle()
    interact(update_ydata, string = selectionset )

manipulate_plot(fig, mydata, mydata.dtype.names[1:])
```

The notebooks runs without error. I get the figure that I want with a radio button containing the list of the dtype.names. However, if I click on the radio button and select on of the entries (say "-63dBm"), the figures vanishes from the notebook without any error message. 
I have no clue, what I am doing wrong.

I have come to quite a few workarounds already, all of which I don't really like for several reasons. E.g  my first attempt, was very close to the [image browser example][link] from nbviewer.com. But what I dislike about this solution is, that it instantiates a new figure for each interaction with the frontend. 

[link]: http://nbviewer.ipython.org/github/ipython/ipython/blob/2.x/examples/Interactive%20Widgets/Image%20Browser.ipynb

Thanks in advance

**Comment by LightServant** [comment]
I think I have found the issue. It appears my code could not work by construction.

I stumbled upon this on [nbviewer.ipthon.org][id]:
"It's important to realize that widgets are not the same as output, even though they are displayed with display. Widgets are drawn in a special widget area. That area is marked with a close button which allows you to collapse the widgets. Widgets cannot be interleaved with output. Doing so would break the ability to make simple animations using clear_output."

If I put the call of the manipulate_plot method and the plt.show() into different python cells, all works well.

I've got a working setup now, but I find it still a little awkward that output (figure)
 and controls of the output (widgets) are separate instead directly of next to each other.

I'm thinking if it is possible to put the figure as a fixed element inside a container widget, or something?
Any other ideas ?

[id]:http://nbviewer.ipython.org/github/ipython/ipython/blob/2.x/examples/Interactive%20Widgets/Widget%20Basics.ipynb
